### PR TITLE
build: upgrade kafka client

### PR DESCRIPTION
**Pull request by venikkin**
Fixes Kafka client security alert

## Proposed Changes (Mandatory)
  - update Kafka client dependency version
  - remove dead code

### CODE REVIEW

**Comment by venikkin** [comment]
[Fix `neo4j-streams` CVE](https://trello.com/c/CGrjVX5z/490-fix-neo4j-streams-cve)

File: pom.xml [hunk 1]
```diff
@@ -49,10 +49,11 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <java.version>11</java.version>
+        <java.version>11</java.version>
         <kotlin.version>1.9.20</kotlin.version>
         <kotlin.coroutines.version>1.7.3</kotlin.coroutines.version>
         <neo4j.version>4.4.27</neo4j.version>
-        <kafka.version>2.6.3</kafka.version>
+        <kafka.version>3.7.0</kafka.version>
```
Comment (ali-ince) [inline pom.xml:56]: Same as in #619.

**Review by ali-ince — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (venikkin) [reply pom.xml]: 3.4 is the lowest version containing the fix. As we discussed used it.

**Review by ali-ince — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -49,10 +49,12 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <java.version>11</java.version>
+        <java.version>11</java.version>
```
Comment (fbiville) [inline pom.xml:52]: duplicate?

Comment (venikkin) [reply pom.xml]: Thanks for spotting it. Fixed.

**Pull request merged**